go-controller/pkg/node/gateway_init.go

```diff
@@ -488,22 +502,22 @@ func interfaceForEXGW(intfName string) string {
 	return intfName
 }
 
-func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP) error {
+func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP, nodeAnnotator kube.Annotator) error {
 	// A DPU host gateway is complementary to the shared gateway running
 	// on the DPU embedded CPU. it performs some initializations and
 	// watch on services for iptable rule updates and run a loadBalancerHealth checker
 	// Note: all K8s Node related annotations are handled from DPU.
 	klog.Info("Initializing Shared Gateway Functionality on DPU host")
 	var err error
+	var gwIntf, gatewayIntf string
+	gatewayIntfTemp := config.Gateway.Interface
 
-	// Force gateway interface to be the interface associated with kubeNodeIP
-	gwIntf, err := getInterfaceByIP(kubeNodeIP)
+	gwIntf, err = getInterfaceByIP(kubeNodeIP)
 	if err != nil {
 		return err
 	}
 	config.Gateway.Interface = gwIntf
-
-	_, gatewayIntf, err := getGatewayNextHops()
+	_, gatewayIntf, err = getGatewayNextHops()
```

Comment:
I am thinking if this is the best way to get interface name... especially considering that we need to reset config variable for that
do you execute this part of the function https://github.com/ovn-kubernetes/ovn-kubernetes/blob/4de8fd5f958956874bce147e01f11bd0049abcd9/go-controller/pkg/node/gateway_init.go#L161 or something else?